### DISCUSSION TITLE
Potential bug: sls deploy works in CLI but not in CI/CD

### DISCUSSION THREAD

**Discussion by stevemandl** [UnitedIncome/serverless-python-requirements#739] [Q&A]
Not sure if I'm doing something wrong, or if this is a bug.
I have a project with both Nodejs and python functions. I can deploy them fine from the CLI (WSL python 3.8 node 16)
I have the service connected to github and it's configured to deploy on pushes to the dev branch. When I trigger CI with a push, it fails. I have tried using python3.8, 3.9, dockerizePip and various docker images, none work. Here is the deploy part of the build log:

Build step: serverless deploy --stage dev --region us-east-1 --force --org sjm34 --app meter-anomaly-cron --verbose 
Deploying meter-anomaly-cron to stage dev (us-east-1, "emcs-cec" provider)
Compiling with Typescript...
Using local tsconfig.json - tsconfig.json
Warning: "rootDir" from local tsconfig.json is overriden
Typescript compiled.
Generating requirements.txt from Pipfile
× Stack meter-anomaly-cron-dev failed to deploy (42s)
Environment: linux, node 16.15.0, framework 3.22.0 (local), plugin 6.2.2, SDK 4.3.2
Credentials: Serverless Dashboard, "emcs-cec" provider (https://app.serverless.com/sjm34/apps/meter-anomaly-cron/meter-anomaly-cron/dev/us-east-1/providers)
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues
Error:
Error: spawn pipenv ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
Build step failed: serverless deploy --stage dev --region us-east-1 --force --org sjm34 --app meter-anomaly-cron --verbose

**Comment by pgrzesik** [comment]
Hey @stevemandl, thanks for reporting and sorry you've run into trouble. The error suggests that you don't have `pipenv` installed on CI and you seem to be using that for packaging. Is that correct?

**Reply by stevemandl** [reply]
You are probably correct. I'm not sure how to install pipenv in the CI build environment though. The service settings in the sls dashboard don't have many options. The CI/CD build settings only include the region to deploy to and how to link branch/stage. Is there a way to specify an build environment with pipenv?

**Reply by pgrzesik** [reply]
Hey @stevemandl - the best bet would be probably to reach out to support at Serverless Dashboard. They might provide more insights on that
